sources/pmwebapi_test.go

```diff
@@ -0,0 +1,98 @@
+package sources
+
+import (
+	"testing"
+	"encoding/json"
+)
+
+type AutoGenerated struct {
+	Args struct {
+	} `json:"args"`
+	Headers struct {
+		Accept string `json:"Accept"`
+		AcceptEncoding string `json:"Accept-Encoding"`
+		AcceptLanguage string `json:"Accept-Language"`
+		CacheControl string `json:"Cache-Control"`
+		Connection string `json:"Connection"`
+		Host string `json:"Host"`
+		UpgradeInsecureRequests string `json:"Upgrade-Insecure-Requests"`
+		UserAgent string `json:"User-Agent"`
+		XBluecoatVia string `json:"X-Bluecoat-Via"`
+	} `json:"headers"`
+	Origin string `json:"origin"`
+	URL string `json:"url"`
+}
+
+func TestGetRequest(t *testing.T) {
+	testString := "http://httpbin.org/get"
+	response := getRequest(testString)
+	if response == nil {
+		t.Fatal("retrieved an unexpected result")
+	}
+	A := &AutoGenerated{}
+	err := json.Unmarshal(response, A)
+	if err != nil {
+		t.Fatal("fail")
+	}
+	if A.Headers.Host != "httpbin.org" {
+		t.Fatal("fail")
+	}
+}
+
+func TestUnmarshal(t *testing.T) {
+	testUrl := "http://httpbin.org/get"
+	testBody := getRequest(testUrl)
+	testMap := &AutoGenerated{}
+	unmarshal([]byte(testBody),testMap)	
+	if testMap.Headers.Host != "httpbin.org" {
+		t.Fatal("faIL")
+	}
+	if testMap.Origin == "12kk" {
```

Comment:
Why are you checking just for "12kk"? There could be an infinite number of strings it could be that are incorrect outside of "12kk". Might be better to try and parse an IP address - if it's not a valid IP address, fail. Then again, even that might not be a solid test since you could get a valid yet unexpected IP address... Maybe query a more deterministic field instead of this one?